chatkit-core/src/main/kotlin/com/pusher/chatkit/CurrentUser.kt

```diff
@@ -233,45 +348,90 @@ class CurrentUser(
             callback: (Result<Int, Error>) -> Unit
     ) = makeCallback({ syncCurrentUser.sendMessage(roomId, messageText, attachment) }, callback)
 
+    /**
+     * Send a [messageText] to the [roomId]. The [callback] will be called with a [Result] when the
+     * operation is complete. If the operation was successful you will receive an [Int] which
+     * contains the message id, or an [Error] informing you of what went wrong.
+     */
     fun sendSimpleMessage(
             roomId: String,
             messageText: String,
             callback: (Result<Int, Error>) -> Unit
     ) = makeCallback({ syncCurrentUser.sendSimpleMessage(roomId, messageText) }, callback)
 
+    /**
+     * @see [sendSimpleMessage]
+     */
     fun sendSimpleMessage(
             room: Room,
             messageText: String,
             callback: (Result<Int, Error>) -> Unit
     ) = makeCallback({ syncCurrentUser.sendSimpleMessage(room, messageText) }, callback)
 
+    /**
+     * Sends some message [parts] to the [room]. For more information on what the [parts] can be check
+     * out the [docs][https://pusher.com/docs/chatkit/reference/android].
+     * The [callback] will be called with a [Result] when the operation is complete. If the
+     * operation was successful you will receive an [Int] which contains the message id, or an
+     * [Error] informing you of what went wrong.
+     */
     fun sendMultipartMessage(
             room: Room,
             parts: List<NewPart>,
             callback: (Result<Int, Error>) -> Unit
     ) = makeCallback({ syncCurrentUser.sendMultipartMessage(room, parts) }, callback)
 
+    /**
+     * @see [sendMultipartMessage]
+     */
     fun sendMultipartMessage(
             roomId: String,
             parts: List<NewPart>,
             callback: (Result<Int, Error>) -> Unit
     ) = makeCallback({ syncCurrentUser.sendMultipartMessage(roomId, parts) }, callback)
 
+    /**
+     * Sets that the current user is typing in the [room]. You can send as many of these as you like,
+     * the SDK will handle rate limiting the requests. The [callback] will be called with a [Result]
+     * when the operation is complete. If the operation was successful you will receive a [Unit], or
+     * an [Error] informing you of what went wrong.
+     */
     fun isTypingIn(room: Room, callback: (Result<Unit, Error>) -> Unit) =
             makeCallback({ syncCurrentUser.isTypingIn(room) }, callback)
 
+    /**
+     * @see [isTypingIn]
+     */
     fun isTypingIn(roomId: String, callback: (Result<Unit, Error>) -> Unit) =
             makeCallback({ syncCurrentUser.isTypingIn(roomId) }, callback)
 
+    /**
+     * Get a list of the public rooms that you are not a member of yet. The [callback] will be called
+     * with a [Result] when the operation is complete. If the operation was successful you will receive
+     * a List<Room>, or an [Error] informing you of what went wrong.
+     */
```

Comment:
Any method returning a `Room` (this applies to others above as well) should specify the expectations on which optional fields will be populated, e.g. `memberUserIds`, `unreadCount`, `lastMessageId` and co. as this is often confusing.

Reply:
that's worth pointing out! `lastMessageId` won't be coming until we update the API so I've written this to be less exact:
` Get a list of the public rooms that you are not a member of yet. Not all fields in the room objects will be populated until you join and subscribe to them e.g. you will not be able to see the `memberUserIds`, until you subscribe to a room.` - what do you think? I could be more exact with which properties are expected not to be populated?

Reply:
I think in this case we can be quite precise - none of `memberUserIds`, `unreadCount` or `lastMessageId` will be populated on the returned rooms, because all of those things require you to be a member, and this returns only rooms you are not a member of.

However, there may be other calls which return room entities where we should also be explicit if we can.

Reply:
I've updated to say: `Get a list of the public rooms that you are not a member of yet. The memberUserIds, unreadCount, and lastMessageId will not be populated until you join and subscribe to the room.`.

I think the only other place we need to update is the `rooms` property - which we've not currently documented.

Reply:
Excellent. The wording `will not be populated until you join and subscribe to the room` does introduce some potential ambiguity though: the `Room` instances you get back from this call will not change to include those props if you do join and subscribe (though you may then receive other instances of `Room` for the same room which do include them from other methods / properties of the SDK.

Perhaps better to say `The memberUserIds, unreadCount, and lastMessageId will not be populated because you do not have access to these as a non-member of the room` or similar.

Reply:
I've expanded to include both your feedback comments:
```
* The memberUserIds, unreadCount, and lastMessageId will not be populated as you are not yet subscribed to these rooms.
     * The rooms returned here will not be updated to populate those properties when you do subscribe, 
     * instead you will receive a room object in the subscription event which will contain all these properties.
```

Reply:
:+1: